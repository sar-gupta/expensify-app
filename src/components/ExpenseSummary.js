import React from 'react';
import { connect } from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = ({ expenseCount, expensesTotal }) => (
  <div>
    <h1>Viewing {expenseCount} {expenseCount===1?'expense':'expenses'} totalling {numeral(expensesTotal/100).format('$0,0.00')}</h1>
  </div>
);

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpenseSummary);