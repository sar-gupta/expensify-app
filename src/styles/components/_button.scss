.button {
  padding: $s-size;
  color: white;
  background: $blue;
  border: none;
  font-size: $font-size-large;  
  font-weight: 300;
  text-decoration:none;
  display: inline-block;
  line-height: 1;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888;
}