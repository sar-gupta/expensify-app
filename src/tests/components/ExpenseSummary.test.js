import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('should render single expense summary correctly', () => {
  const expenses = {
    expenseCount: 1,
    expensesTotal: 250
  };
  const wrapper = shallow(<ExpenseSummary {...expenses}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render multiple expenses summary correctly', () => {
  const expenses = {
    expenseCount: 2,
    expensesTotal: 2500
  };
  const wrapper = shallow(<ExpenseSummary {...expenses}/>);
  expect(wrapper).toMatchSnapshot();
});