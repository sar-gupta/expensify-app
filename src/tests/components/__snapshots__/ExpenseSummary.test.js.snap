// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render multiple expenses summary correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        2
      </span>
       
      expenses
       totalling 
      <span>
        $25.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render single expense summary correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $2.50
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
